### DISCUSSION TITLE
Contribution Guide

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/runtime#73887] [Q&A] [answered]
That's Contributing GUIDE is very Confused me.
how we can to contribute dotnet Repo's and make PR's!?
can described this for me!?

**Comment by huoyaoyuan** [comment, marked as answer]
You can open PR for things you want to change. Some rules to remember in short:
- Fix for simple typos can be submitted directly without any concern.
- Changing the shape of anything public (public members of public types) would require an issue approved in advance. See [api review process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md).
- Bug fixes should have tests added.

If you are not sure about your idea about changing, you can open an issue to discuss it first.

**Reply by danmoseley** [reply]
What is confusing to you or missing that you expected @sajjadarashhh ? We would welcome a PR to improve it.

**Reply by sajjadarashhh** [reply]
Im dont know how can i help in performance issues or etc...
this happens when i dont know wheres the documentation
And how wasm compiler works

**Reply by sajjadarashhh** [reply]
Sorry for dely to asking you!

**Reply by huoyaoyuan** [reply]
You can see the guidance in https://github.com/dotnet/performance and https://github.com/dotnet/BenchmarkDotNet to know how to measure performance.
